File: src/main/java/net/guizhanss/ultimategenerators2/core/recipes/SingleMachineRecipe.java
package net.guizhanss.ultimategenerators2.core.recipes;

import org.bukkit.inventory.ItemStack;

/**
 * A {@link MachineRecipe} but only has one input and one output.
 */
public class SingleMachineRecipe extends SingleOutputMachineRecipe {
    public SingleMachineRecipe(int ticks, ItemStack input, ItemStack output) {
        super(ticks, new ItemStack[] {input}, output);
    }
}

File: src/main/kotlin/net/guizhanss/ultimategenerators2/core/recipes/SingleMachineRecipe.kt
package net.guizhanss.ultimategenerators2.core.recipes

import org.bukkit.inventory.ItemStack

/**
 * A [MachineRecipe] but only has one input and one output.
 */
class SingleMachineRecipe(ticks: Int, input: ItemStack, output: ItemStack) :
    SingleOutputMachineRecipe(ticks, arrayOf(input), output)
